Library/Homebrew/dev-cmd/ruby.rb

```diff
@@ -13,21 +13,26 @@ def ruby_args
         Run a Ruby instance with Homebrew's libraries loaded, e.g.
         `brew ruby -e "puts :gcc.f.deps"` or `brew ruby script.rb`.
       EOS
-      switch "-r",
-             description: "Load a library using `require`."
-      switch "-e",
-             description: "Execute the given text string as a script."
+      flag "-r=",
+           description: "Load a library using `require`."
+      flag "-e=",
+           description: "Execute the given text string as a script."
     end
   end
 
   def ruby
-    ruby_args.parse
+    args = ruby_args.parse
+
+    ruby_sys_ags = []
```

Comment:
```suggestion
    ruby_sys_args = []
```

(and same elsewhere)